app/lib/bk/compat/models/pipeline.rb

```diff
@@ -29,7 +29,8 @@ def simplify_steps
 
       def to_h
         {}.tap do |h|
-          h[:env] = @env unless @env.empty?
+          # Conditionally include the env section only if it contains valid environment variables
+          h[:env] = @env unless @env.empty? || @env.keys.all? { |k| k.to_s.strip.empty? }
```

Comment:
I don't think that this change should be necessary, the checks for that should be done before adding the values to the hash